### DISCUSSION TITLE
Micrometer Observation Question

### DISCUSSION THREAD

**Discussion by murphy-25** [spring-projects/spring-kafka#3389] [General]
Hi,

I'm upgrading some services to the latest versions of spring-kafka. We currently setup a MicrometerConsumerListener to enable Kafka metrics like below.
`
defaultConsumerFactory.addListener(new MicrometerConsumerListener<>(registry));
`

In the new version of spring-kafka we are now also setting up a consumer with observation enabled like so:
`
ConcurrentKafkaListenerContainerFactory<String, String> factory = new ConcurrentKafkaListenerContainerFactory<>();
        factory.setContainerCustomizer(container -> {
            container.getContainerProperties().setAuthExceptionRetryInterval(Duration.ofSeconds(30L));
            container.getContainerProperties().setObservationEnabled(true);
        });
`

Am I right in saying that setting observation enabled to true should mean specifying the MicrometerConsumerListener is now redundant?

**Comment by artembilan** [comment]
> Am I right in saying that setting observation enabled to true should mean specifying the `MicrometerConsumerListener` is now redundant?

No, that not.

The observation does this for each record:
```
			return observation.observe(() -> {
				try {
					invokeOnMessage(cRecord);
					successTimer(sample, cRecord);
					recordInterceptAfter(cRecord, null);
				}
```

That  `MicrometerConsumerListener` deals with the `KafkaClientMetrics`, which is based on the `kafkaConsumer::metrics`.
I think they would just supplement each other according to their docs: 

https://docs.confluent.io/platform/current/kafka/monitoring.html#kafka-monitoring-metrics-consumer
https://docs.spring.io/spring-kafka/reference/appendix/micrometer.html
